Fix librealsense2 build on ros2 humble/iron

akssri-sony opened the pull request.

wentasah commented on distros/librealsense2-dont-build-nlohmann-json.patch:
```diff
@@ -0,0 +1,17 @@
+--- 1,6 ----
+  string(REPLACE ${PROJECT_SOURCE_DIR}/ "" _rel_path ${CMAKE_CURRENT_LIST_DIR})
+  
+! # include(CMake/external_json.cmake)
```
We prefer not to have patches here, but instead fetch them from upstream PRs or development commits. This is, of course, not an upstreamable patch. If we want it here, I'd prefer so called unified diff format, which is more readable. It can be produced with `git diff --no-index` or `diff -u`. However, it this case, we can do without the patch. We can use [`substituteInPlace`](https://nixos.org/manual/nixpkgs/unstable/#fun-substituteInPlace) in the override:

```
...
postPatch = postPatch + ''
  substituteInPlace third-party/CMakeLists.txt --replace-fail 'include(CMake/external_json.cmake)' ""
'';
```

akssri-sony replied:
sg; I've replaced the patch with the postPatch.

wentasah reviewed (changes requested):
Hi, as noted in the comments below, it would be better to move all the needed changes to `overrides.nix`. We do similar things with other packages so these files are full of examples how to do the changes.

akssri-sony commented on distros/ros2-overlay.nix:
```diff
@@ -66,14 +66,18 @@ rosSelf: rosSuper: with rosSelf.lib; {
+        (self.fetchpatch {
+          url = "https://github.com/IntelRealSense/librealsense/commit/847b74d3dcade2842ba138f321474159315ab8c2.patch";
+          hash = "sha256-zaW8HG8rfsApI5S/3x+x9Fx8xhyTIPNn/fJVFtkmlEA=";
+        })
```
sg; I can't build foxy for some reason, but I've kept the postPatch restricted to ">= 2.55.1".

akssri-sony commented on distros/humble/librealsense2/default.nix:
```diff
@@ -2,7 +2,7 @@
 # Distributed under the terms of the BSD license
 
-{ lib, buildRosPackage, fetchurl, cmake, git, glfw3, libGL, libGLU, libusb1, openssl, pkg-config, udev, xorg }:
+{ lib, buildRosPackage, fetchurl, cmake, git, glfw3, libGL, libGLU, libusb1, openssl, pkg-config, nlohmann_json, udev, xorg }:
```
done; moved everything to ros2-overlay.nix.

akssri-sony commented:
@wentasah Thanks for the review! I've fixed the issues raised, and moved all the build-modifications into ros2-overlay.nix.
I couldn't test foxy, but it doesn't touch that package now (other than removing the stale patch).
